packages/ods/src/components/pagination/src/components/ods-pagination/ods-pagination.tsx

```diff
@@ -0,0 +1,338 @@
+    if (!this.totalItems) {
+      return this.totalPages;
+    }
+
+    return Math.ceil(this.totalItems / (itemPerPage || 1));
+  }
+
+  private updatePageList(): void {
+    this.pageList = createPageList(this.actualTotalPages, this.current);
+  }
+
+  @Listen('odsChange')
+  odsValueChangeHandler(event: CustomEvent<OdsSelectEventChangeDetail>): void {
+    event.preventDefault();
+    event.stopPropagation();
+
+    const { value } = event.detail;
+
+    if (value) {
+      this.itemPerPage = typeof value === 'number' ? value : parseInt(value, 10);
+    }
+  }
+
+  @Watch('labelTooltipNext')
+  @Watch('labelTooltipPrevious')
+  updateLabelTooltip(): void {
+    forceUpdate(this.el);
+  }
+
+  @Watch('current')
+  async onCurrentChange(current: number, oldCurrent?: number): Promise<void> {
+    this.updatePageList();
+    this.emitChange(current, oldCurrent);
+  }
+
+  @Watch('totalPages')
+  async onTotalPagesChange(): Promise<void> {
+    if (!this.totalItems) {
+      this.actualTotalPages = this.totalPages;
+    }
+    this.updatePageList();
+  }
+
+  @Watch('itemPerPage')
+  async onItemPerPageChange(): Promise<void> {
+    if (!this.isFirstLoad) {
+      await this.updatePagination();
+
+      this.odsPaginationItemPerPageChanged.emit({
+        current: this.itemPerPage,
+        currentPage: this.current,
+        totalPages: this.actualTotalPages,
+      });
+    }
+  }
+
+  @Watch('totalItems')
+  async onTotalItemsChange(): Promise<void> {
+    await this.updatePagination();
+  }
+
+  @Method()
+  async setPageIndex(current: number): Promise<void> {
+    this.current = current;
+  }
+
+  @Method()
+  async getCurrentPage(): Promise<number> {
+    return this.current;
+  }
+
+  private emitChange(current: number, oldCurrent?: number): void {
+    this.odsPaginationChanged.emit({
+      current: current,
+      itemPerPage: this.itemPerPage,
+      oldCurrent: oldCurrent,
+    });
+  }
+
+  private async updatePagination(): Promise<void> {
+    this.actualTotalPages = this.computeActualTotalPages(this.itemPerPage);
+
+    if (this.current === 1) {
+      // If current is already 1 we don't want to emit a change event
+      this.updatePageList();
+    } else {
+      await this.setPageIndex(1);
+    }
+  }
+
+  handlePreviousClick(page: number): void {
+    this.setPageIndex(page - 1);
+  }
+
+  handleNextClick(page: number): void {
+    this.setPageIndex(page + 1);
+  }
+
+  handlePageClick(page: number): void {
+    this.setPageIndex(page);
+  }
+
+  handlePreviousKeyDown(event: KeyboardEvent, page: number): void {
+    if (this.current > 1) {
+      this.onKeyDown(event, page - 1);
+    }
+  }
+
+  handleNextKeyDown(event: KeyboardEvent, page: number): void {
+    if (this.current < this.pageList.length) {
+      this.onKeyDown(event, page + 1);
+    }
+  }
+
+  handlePageKeyDown(event: KeyboardEvent, page: number): void {
+    this.onKeyDown(event, page);
+  }
+
+  onKeyDown(event: KeyboardEvent, page: number): void {
```

Comment:
if possible use `onKeyUp` (and manage the Space prevention to avoid scroll on key down)